packages/blocks/src/api/registration.js

```diff
@@ -186,11 +186,12 @@ export function registerBlockType( name, settings ) {
 		supports: {},
 		styles: [],
 		save: () => null,
+		...settings,
+		// Use the existing definition from the server as a source of truth.
```

Comment:
Any context about the change here? Any idea how impactful it is

Reply:
If I recall properly, we discussed the fact that the current order make sense for instance if you want to provide a string icon on the server and an svg override in  the client

Reply:
I didn't think about the `icon`, but I will add a description and therefore my motivation in the description.

I should wait with review requests, you were too fast and I got distracted with something else 😅

Reply:
I updated the description. Can you think about anything else that we would want to override on the client that can't be encoded in `block.json`?

Reply:
I think I still don't really understand why we do want to switch :) 

It's true that we want most registration for blocks to be server-side first but I'm not sure I understand the reasoning for allowing it to be overridable in the client since it's what loads last.

Reply:
One thing I'm worried about is the case when you try to load the same data on the server and client using shared `block.json` file. On the server it's processed with `register_block_type_from_metadata` so some fields are wrapped with i18n functions and so on. There's also several ways to filter the metadata registered. There is no such thing on the client that operates on the `block.json`so it might create some friction. Anyway, it's something that might need only some tweaks in the documentation and a dev note published. This is what I'm trying to clarify.